dot/state/epoch.go

```diff
@@ -134,7 +134,7 @@ func NewEpochStateFromGenesis(db chaindb.Database, blockState *BlockState,
 }
 
 // NewEpochState returns a new EpochState
-func NewEpochState(db chaindb.Database, blockState *BlockState) (*EpochState, error) {
+func NewEpochState(db *chaindb.BadgerDB, blockState *BlockState) (*EpochState, error) {
```

Comment:
This is left as an implementation on purpose, since it should be changed by issue #2981